File: hexameter-core/src/test/java/org/codetome/hexameter/api/HexagonOrientationTest.java
package org.codetome.hexameter.api;

import static junit.framework.Assert.assertEquals;
import static org.codetome.hexameter.api.HexagonOrientation.FLAT_TOP;
import static org.codetome.hexameter.api.HexagonOrientation.POINTY_TOP;
import static org.codetome.hexameter.testutils.TestUtils.superficialEnumCodeCoverage;

import org.codetome.hexameter.api.HexagonOrientation;
import org.junit.Test;

public class HexagonOrientationTest {

	@Test
	public void testEnum() {
		superficialEnumCodeCoverage(HexagonOrientation.class);
	}

	@Test
	public void shouldProperlyCalculateFlatCoordinateOffset() {
		assertEquals(0.0f, FLAT_TOP.getCoordinateOffset());
	}

	@Test
	public void shouldProperlyCalculatePointyCoordinateOffset() {
		assertEquals(0.5f, POINTY_TOP.getCoordinateOffset());
	}
}

File: mixite.core/core/src/test/kotlin/org/hexworks/mixite/core/api/HexagonOrientationTest.kt
package org.hexworks.mixite.core.api

import org.hexworks.mixite.core.api.HexagonOrientation.*
import kotlin.test.Test
import kotlin.test.assertEquals

class HexagonOrientationTest {

    @Test
    fun shouldProperlyCalculateFlatCoordinateOffset() {
        assertEquals(expected = 0.0f, actual = FLAT_TOP.coordinateOffset)
    }

    @Test
    fun shouldProperlyCalculatePointyCoordinateOffset() {
        assertEquals(expected = 0.5f, actual = POINTY_TOP.coordinateOffset)
    }
}
